gradoop-flink/src/test/java/org/gradoop/flink/model/impl/operators/kmeans/KMeansTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2014 - 2021 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.kmeans;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.gradoop.common.model.api.entities.Vertex;
+import org.gradoop.flink.model.GradoopFlinkTestBase;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.operators.kmeans.util.*;
+import org.gradoop.flink.model.impl.operators.kmeans.functions.*;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class KMeansTest<V extends Vertex> extends GradoopFlinkTestBase {
+
+  private List<Tuple2<Centroid, Point>> expectedNearestCenter = new ArrayList<>();
+  private List<Tuple3<Integer, Point, Long>> expectedNearestCenterWithAppendedCounter = new ArrayList<>();
+  private List<Tuple3<Integer, Point, Long>> expectedCentroidsWithAccumulatedPoints = new ArrayList<>();
+  private List<Centroid> expectedAverageCentroids = new ArrayList<>();
+  String inputGdlGraph;
+
+  // init execution environment
+  ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+  DataSet<Point> points = env
+    .fromElements(new Point(20.0, 10.0), new Point(25.0, 15.0), new Point(30.0, 20.0), new Point(35.0, 25.0),
+      new Point(40.0, 30.0), new Point(10.0, 10.0), new Point(10.0, 30.0), new Point(20.0, 40.0),
+      new Point(5.0, 10.0));
+
+  DataSet<Centroid> centroids = env
+    .fromElements(new Centroid(1, new Point(10.0, 15.0)), new Centroid(2, new Point(20.0, 20.0)),
+      new Centroid(3, new Point(25.0, 30.0)));
+
+  @Before
```

Comment:
All this stuff will be executed before _each_ test function in this class. Try to use static @BeforeClass to avoid multiple unnecessary executions.

If you are using a list with expected values just in a single test function, in does not need to be a class variable, but can be created in the test function.

Reply:
Done